### ISSUE TITLE
[Umbrella] Can't start EC2 instance on EC2-Classic account

### ISSUE THREAD

**Issue by Baguage** [cooperative-computing-lab/cctools#1070]
Trying to use umbrella from master to run openmalaria on EC2 - and it fails with error below
**The AssociatePublicIpAddress parameter is only supported for VPC launches.**

[avyushko@stockpile umbrella-om-spec]$ umbrella --spec openmalaria.umbrella --localdir /tmp/umbrella_test/ --output "/tmp/ctsout.txt=/tmp/umbrella_test/ec2_openmalaria/ctsout.txt, /tmp/output.txt=/tmp/umbrella_test/ec2_openmalaria/output.txt" --sandbox_mode ec2 --ec2_sshkey ~/.ssh/id_rsa --ec2_key feb272015 --ec2_instance_type m3.medium --ec2_log umbrella.log.ec2 --log umbrella.log run

dump the metadata information from the umbrella spec to /tmp/umbrella_test/tmp37tGKl/meta.json
Starting validating the spec file ....
Finish validating the spec file successfully!
Checking the validity of the umbrella specification ...
Deciding the AMI according to the umbrella specification ...
Starting an Amazon EC2 instance ...
Client.InvalidParameter: The AssociatePublicIpAddress parameter is only supported for VPC launches. (Service: AmazonEC2; Status Code: 400; Error Code: InvalidParameter; Request ID: b2980c9b-5ff3-4c32-937e-271b9b306295)
cleanup temporary dir:  /tmp/umbrella_test/tmp37tGKl
'ec2-run-instances ami-0b06483b -t m3.medium -k feb272015  --associate-public-ip-address true' fails with the return code of 1, 
stdout: , 
stderr: None

**Comment by hmeng-19** [comment]
I believe the problem is the `--ec2_key` option, which should be your amazon key-pair. `feb272015` is my own amazon key-pair. You may need to first create your own key-pair:
http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-key-pairs.html

**Comment by hmeng-19** [comment]
Your `--ec2_sshkey` option also looks a little skeptical, which should be the path of the private key file to use when connecting to an Amazon EC2 instance.

**Comment by Baguage** [comment]
Ok, I have corrected that, but it does not seem to solve my problem
```
[avyushko@stockpile umbrella-om-spec]$ umbrella --spec openmalaria.umbrella --localdir /tmp/umbrella_test/ --output "/tmp/ctsout.txt=/tmp/umbrella_test/ec2_openmalaria/ctsout.txt, /tmp/output.txt=/tmp/umbrella_test/ec2_openmalaria/output.txt" --sandbox_mode ec2 --ec2_sshkey umbrella.pem --ec2_key umbrella --ec2_instance_type m3.medium --ec2_log umbrella.log.ec2 --log umbrella.log run
dump the metadata information from the umbrella spec to /tmp/umbrella_test/tmphOBxjV/meta.json
Starting validating the spec file ....
Finish validating the spec file successfully!
Checking the validity of the umbrella specification ...
Deciding the AMI according to the umbrella specification ...
Starting an Amazon EC2 instance ...
Client.InvalidParameter: The AssociatePublicIpAddress parameter is only supported for VPC launches. (Service: AmazonEC2; Status Code: 400; Error Code: InvalidParameter; Request ID: 4aed3535-1ddd-47b6-9f5c-13b82b10399e)
cleanup temporary dir:  /tmp/umbrella_test/tmphOBxjV
`ec2-run-instances ami-0b06483b -t m3.medium -k umbrella  --associate-public-ip-address true` fails with the return code of 1, 
stdout: , 
stderr: None
```

One piece of information - I'm using EC2-Classic account for this test (my personal account), and I do not have access to EC2-VPC. I have a sneaking suspicion that you can't assigned a public IP to a instance in EC2-Classic account.

**Comment by hmeng-19** [comment]
Instances from the EC2-Classic Platform by default should have a public IP address:
     http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-vpc.html

My EC2 account by default run instances s on the EC2-VPC platform. 

Let us leave this problem for this Wed's meeting, we can troubleshoot together.

**Comment by Baguage** [comment]
Sounds like a plan!

**Comment by hmeng-19** [comment]
http://docs.aws.amazon.com/AmazonVPC/latest/UserGuide/default-vpc.html#default-vpc-availability

**Comment by hmeng-19** [comment]
http://docs.aws.amazon.com/AmazonVPC/latest/UserGuide/default-vpc.html#detecting-platform

**Comment by hmeng-19** [comment]
http://docs.aws.amazon.com/AWSEC2/latest/CommandLineReference/ec2-cli-launch-instance.html#launching-instances

**Comment by Baguage** [comment]
For the record, the root cause of the problem is my account. It was created many years ago, and I did not migrate to new EC-VPC platform (still use EC-Classic). As discussed with Haiyan, umbrella should probably detect those accounts and give a warning to user. In the mean time, I will create a new account and try umbrella with EC-VPC account.

**Comment by hmeng-19** [comment]
This problem finally turns out to be the resource scope of the Amazon EC2:
http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/resources.html

Fixed by #1105 .
